src/utils/upload.js

```diff
@@ -23,15 +25,17 @@ export default class Upload {
     params = {},
     chunkLen = UPLOAD_CHUNK_SIZE,
   } = {}) {
+    super(file, {
+      $rest, parent, progress,
+    });
     Object.assign(this, {
-      $rest, file, params, parent, progress, chunkLen, upload: null, offset: 0, behavior: null,
+      params, chunkLen, upload: null, offset: 0, behavior: null,
     });
   }
 
   async _sendChunks() {
     const onUploadProgress = e => this.progress({
       current: this.offset + e.loaded,
-      total: this.file.size,
```

Comment:
What's the rationale for not passing this to the callback? I realize the receiver almost certainly has access to the total, but it seems helpful (or at least harmless) to also include it here.

Reply:
I first noticed this is because the total is not being used in Upload.vue, then I realize Upload.vue is [aware of the size itself](https://github.com/girder/girder_web_components/blob/master/src/components/Upload.vue#L88). Because it is Upload.vue that passed the file to upload.js to upload, It's natural that Upload.vue is aware of things like file size, name. I think file size is something given to upload.js, not something it is able to change or responsible to report.

Reply:
We should also consider the fact that the upload class could be used in other contexts besides the Upload.vue component. Callers will likely always know the size, but it makes for a nice API to include the total IMO, so you can have a callback like:

```javascript
({ current, total }) => { this.percentDone = 100 * current / total; }
```

I could be in the minority here, though. What do others think? @subdavis @brianhelba

Reply:
I haven't tried to do anything with that callback, but I agree that the API is nice to have.  Even when `Upload.vue` is the source of the data, the handler could be somewhere else.   I wouldn't want to reach into that component's private data to get this info.

Reply:
Make sense, but can I take the liberty to change the name to size? Because size is more explicit than total and the word total has been used in Upload.vue for real total.

Reply:
Fine by me.